internal/controllers/app_controller.go

```diff
@@ -200,6 +210,16 @@ func (r *AppReconciler) reconcile(ctx context.Context, app *ketchv1.App) reconci
 
 	// check for canary deployment
 	if app.Spec.Canary.Active {
+		// retry until all pods for canary deployment comes to running state.
+		if err := canaryPodStatus(r.Client, app); err != nil {
+			return reconcileResult{
+				status:     v1.ConditionFalse,
+				message:    fmt.Sprintf("canary upgrade failed: %v", err),
```

Comment:
Why `upgrade`? Do we upgade anything here? Sounds like we tried to upgrade canary itself, but failed.

Reply:
I would like to clarify that we update the canary weights after step interval. I agree with you, I would use `update` instead. Thanks!